src/ImageSearchModal.jsx

```diff
@@ -9,244 +9,214 @@ import { Radio } from "@patternfly/react-core/dist/esm/components/Radio";
-                                        <DataListItemCells
-                                                  dataListCells={[
-                                                      <DataListCell key="primary content">
-                                                          <span className='image-name'>{image.Name}</span>
-                                                      </DataListCell>,
-                                                      <DataListCell key="secondary content">
-                                                          <span className='image-description'>{image.Description}</span>
-                                                      </DataListCell>
-                                                  ]}
-                                        />
-                                    </DataListItemRow>
-                                </DataListItem>
-                            );
-                        })}
-                    </DataList>}
-                </>}
-            </>
-        );
-
-        return (
-            <Modal isOpen className="podman-search"
-                   position="top" variant="large"
-                   onClose={Dialogs.close}
-                   title={_("Search for an image")}
-                   footer={<>
-                       <Form isHorizontal className="image-search-tag-form">
-                           <FormGroup fieldId="image-search-tag" label={_("Tag")}>
-                               <TextInput className="image-tag-entry"
-                                      id="image-search-tag"
-                                      type='text'
-                                      placeholder="latest"
-                                      value={this.state.imageTag || ''}
-                                      onChange={(_event, value) => this.onValueChanged('imageTag', value)} />
-                           </FormGroup>
-                       </Form>
-                       <Button variant='primary' isDisabled={this.state.selected == undefined} onClick={this.onDownloadClicked}>
-                           {_("Download")}
-                       </Button>
-                       <Button variant='link' className='btn-cancel' onClick={Dialogs.close}>
-                           {_("Cancel")}
-                       </Button>
-                   </>}
-            >
-                {defaultBody}
-            </Modal>
-        );
-    }
-}
+    };
+
+    const onToggleUser = ev => setIsSystem(ev.currentTarget.value === "system");
+    const onDownloadClicked = () => {
+        const selectedImageName = imageList[selected].Name;
+        if (activeConnection)
+            activeConnection.close();
+        Dialogs.close();
+        downloadImage(selectedImageName, imageTag, isSystem);
+    };
+
+    const defaultBody = (
+        <>
+            <Form isHorizontal>
+                {dialogError && <ErrorNotification errorMessage={dialogError} errorDetail={dialogErrorDetail} />}
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1332-20230704-165425-9756b240-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)